lib/cmd/compile/_client-init.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+/**
+ * mount legacy services from _global.js, if any
+ */
+function mountLegacyServices() {
+  Object.keys(window.modules)
+    .filter((key) => typeof key === 'string' && key.match(/\.legacy$/))
+    .forEach((key) => window.require(key));
+}
+
+function tryToMount(fn, el, name) {
+  try {
+    fn(el); // init the controller
+  } catch (e) {
+    const elementTag = el.outerHTML.slice(0, el.outerHTML.indexOf(el.innerHTML));
+
+    console.error(`Error initializing controller for "${name}" on "${elementTag}"`);
+  }
+}
+
+/**
+ * mount client.js component controllers
+ */
+function mountComponentModules() {
+  Object.keys(window.modules)
+    .filter((key) => typeof key === 'string' && key.match(/\.client$/))
+    .forEach((key) => {
+      let controllerFn = window.require(key);
+
+      if (typeof controllerFn === 'function') {
+        const name = key.replace('.client', ''),
+          instancesSelector = `[data-uri*="_components/${name}/"]`,
+          defaultSelector = `[data-uri$="_components${name}"]`,
+          instances = document.querySelectorAll(instancesSelector),
+          defaults = document.querySelectorAll(defaultSelector);
+
+        for (let el of instances) {
```

Comment:
`for (let el of [...instances, ...defaults]) { `
😎 😎 😎

Reply:
spread is cool, but still sorta new (node 8.3.0+, and only has _full_ support in chrome and firefox right now). I'm gonna hold off for now.